internal/goofys.go

```diff
@@ -604,8 +604,8 @@ func (fs *Goofys) LookUpInode(
 	parent := fs.getInodeOrDie(op.Parent)
 	fs.mu.Unlock()
 
-	parent.mu.Lock()
```

Comment:
is there a reason why this order is changed?

Reply:
In function `ForgetInode` lock order is `fs.mu.Lock()` -> `parent.mu.Lock()`
In function `LookUpInode`, if still keep old order  `parent.mu.Lock()` -> `fs.mu.Lock()`, would lead to deadlock.

That is function `ForgetInode` get `fs.mu.Lock()` and wait for  `parent.mu.Lock()`, at same time function `LookUpInode` get `parent.mu.Lock()` and wait for `fs.mu.Lock()`.

Reply:
Same reason for code change in another commit.

I find the deadlock problem when running with only code change in function `ForgetInode`.